HW_1/Task_1.py

```diff
@@ -0,0 +1,327 @@
+
+**Output:**
+
+2'''
+
+
+def distance():
+    print('Задание 1')
+    n = int(input())
+    m = int(input())
+    print((m-1)//n+1)
+
+# distance()
+
+
+'''2. В некоторой школе решили набрать три новых математических класса и оборудовать кабинеты для них новыми партами.  
+За каждой партой может сидеть два учащихся. Известно количество учащихся в каждом из трех классов.  
+Выведите наименьшее число парт, которое нужно приобрести для них.
+
+**Input:**
+
+20
+
+21
+
+22
+
+**Output:**
+
+32'''
+
+
+def kolichestvo_part(a):
+    return (a-1)//2+1
+
+
+def school():
+    print('Задание 2')
+    a = int(input())
+    b = int(input())
+    c = int(input())
+    print(kolichestvo_part(a)+kolichestvo_part(b)+kolichestvo_part(c))
+
+# school()
+
+
+'''3. Требуется вывести все целые степени двойки (т.е. числа вида 2k), не превосходящие числа N.  
+В решении дллжно быть не более 5 строк кода'''
+
+
+def dvoiki():
+    print('Задание 3')
+    N = float(input())
+    k = 0
+    while (2**k) < N:
+        print(k)
+        k += 1
+
+# dvoiki()
+
+
+'''4. Дана последовательность из N целых чисел и число K.  
+Необходимо сдвинуть всю последовательность (сдвиг - циклический) на K элементов вправо, K – положительное число.  
+Не использовать циклы.
+
+Input:   [1, 2, 3, 4, 5] k = 3
+
+Output:  [4, 5, 1, 2, 3]'''
+
+
+def list_4():
+    print('Задание 4')
+    list_1 = list(map(int, input().split(',')))
+    k = int(input())
+    list_1 = list_1[k:]+list_1[:k]
+    print(list_1)
+
+# list_s()
+
+
+'''5. Дан массив, состоящий из целых чисел.  
+Напишите программу, которая подсчитает количество элементов массива, больших предыдущего (элемента с предыдущим номером)
+
+Input: [0, -1, 5, 2, 3]
+Output: 2
+
+Пояснение: (-1 < 5, 2 < 3)'''
+
+
+def list_5():
+    print('Задание 5')
+    list_1 = list(map(int, input().split(',')))
+    count = 0
+    for i in range(len(list_1)-1):
+        if list_1[i] < list_1[i+1]:
+            count += 1
+    print(count)
+
+# list_5()
+
+
+'''6. Напишите программу, которая на вход принимает два числа A и B, и возводит число А в целую степень B с помощью рекурсии.'''
+
+
+def recursiya(a, b):
+    if b == 0:
+        return 1
+    else:
+        return a*recursiya(a, b-1)
+
+
+def recursiya_zadanie():
+    print('Задание 6')
+    a = float(input())
+    b = int(input())
+    print(recursiya(a, b))
+
+# recursiya_zadanie()
+
+
+'''7. Напишите функцию, которая принимает одно число и проверяет, является ли оно простым
+Напоминание: Простое число - это число, которое имеет 2 делителя: 1 и n(само число)
+
+Input: 5
+
+Output: yes'''
+
+
+def simple_number(a):
+    if a != 2 and a % 2 == 0 or a != 3 and a % 3 == 0 or a != 5 and a % 5 == 0 or a != 7 and a % 7 == 0:
+        return False
+    else:
+        count = 0
+        for i in range(1, a-1):
+            if a % i == 0:
+                count += 1
+            if count > 1:
+                break
+        if count > 1:
+            return False
+        else:
+            return True
```

Comment:
Если хоть раз делится, то значит число уже не простое. Зачем переменная count?
```suggestion
        for i in range(2, int(a ** (0.5)) + 1):  # согласно теореме из теория чисел
            if a % i == 0:
                return False
        return True
```
вот и все. Хотя это не самая идеальна реализация